test/apps/iodemo/io_demo.cc

```diff
@@ -587,13 +585,27 @@ class P2pDemoCommon : public UcxContext {
             return _size;
         }
 
+    private:
+        Buffer(void *buffer, size_t size, BufferMemoryPool<Buffer> *pool,
```

Comment:
make this constructor private, since it should not be used directly. only by allocate()